static/js/compose_actions.js

```diff
@@ -292,10 +292,29 @@ export function start(msg_type, opts) {
         return;
     }
     notifications.clear_compose_notifications();
-    expand_compose_box();
 
     opts = fill_in_opts_from_current_narrowed_view(msg_type, opts);
 
+    // Close and block the compose box for the pm-with case, if
+    // the recipient is a deactivated user else,
+    // expand the compose box.
+    if (opts.private_message_recipient !== "") {
+        if (
+            people.verify_non_active_human_id(
+                Number.parseInt(
+                    people.emails_strings_to_user_ids_string(opts.private_message_recipient),
+                    10,
+                ),
+            )
```

Comment:
Un-nesting this `if`, using `&&` should make this code more concise and readable

Reply:
made the change, Thanks